tests/e2e/specs/shopper/filter-products-by-price.test.ts

```diff
@@ -0,0 +1,210 @@
+import {
+	createNewPost,
+	deleteAllTemplates,
+	insertBlock,
+	switchUserToAdmin,
+	publishPost,
+} from '@wordpress/e2e-test-utils';
+import { selectBlockByName } from '@woocommerce/blocks-test-utils';
+import {
+	BASE_URL,
+	goToTemplateEditor,
+	openBlockEditorSettings,
+	saveTemplate,
+	useTheme,
+} from '../../utils';
+
+const block = {
+	name: 'Filter Products by Price',
+	slug: 'woocommerce/price-filter',
+	class: '.wc-block-price-filter',
+	selectors: {
+		editor: {
+			filterButtonToggle: "//label[text()='Filter button']",
+		},
+		frontend: {
+			priceMaxAmount: '.wc-block-price-filter__amount--max',
+			productsList: '.wc-block-grid__products > li',
+			classicProductsList: '.products.columns-3 > li',
+			submitButton: '.wc-block-components-filter-submit-button',
+		},
+	},
+	urlSearchParamWhenFilterIsApplied: '?max_price=1.99',
+	foundProduct: '32GB USB Stick',
+};
+
+const waitForAllProductsBlockLoaded = () =>
+	page.waitForSelector( selectors.frontend.productsList + '.is-loading', {
+		hidden: true,
+	} );
+
+const goToShopPage = () =>
+	page.goto( BASE_URL + '/shop', {
+		waitUntil: 'networkidle0',
+	} );
+
+const { selectors } = block;
+
+const setMaxPrice = async () => {
+	await page.waitForSelector( selectors.frontend.priceMaxAmount );
+	await page.focus( selectors.frontend.priceMaxAmount );
+	await page.click( selectors.frontend.priceMaxAmount, {
+		clickCount: 3,
+	} );
+	await page.keyboard.type( '1.99' );
+	await page.$eval( selectors.frontend.priceMaxAmount, ( el ) =>
+		( el as HTMLElement ).blur()
+	);
+};
+
+describe( `${ block.name } Block`, () => {
+	describe( 'with All Products Block', () => {
+		beforeAll( async () => {
+			await switchUserToAdmin();
+			await createNewPost( {
+				postType: 'post',
+				title: block.name,
+			} );
+
+			await insertBlock( block.name );
+			await insertBlock( 'All Products' );
+			await publishPost();
+
+			const link = await page.evaluate( () =>
+				wp.data.select( 'core/editor' ).getPermalink()
+			);
+			await page.goto( link );
+		} );
+
+		it( 'should render', async () => {
+			await waitForAllProductsBlockLoaded();
+			const products = await page.$$( selectors.frontend.productsList );
+
+			expect( products ).toHaveLength( 5 );
+		} );
+
+		it( 'should show only products that match the filter', async () => {
+			const isRefreshed = jest.fn( () => void 0 );
+			page.on( 'load', isRefreshed );
+			await setMaxPrice();
+			await page.waitForTimeout( 1000 );
```

Comment:
is there a way to replace that with something more meaningful and reliable than time? 

What are we waiting here for? for the `setMaxPrice` to work correctly? can we add something inside that helper to wait for ie. updated element state, idle network or something that makes sense?